betamax: Disable tests

matthiasbeyer opened the pull request:
Closes #15030 
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

This is a proposal. It is a short-term fix for building betamax with python 3. Nevertheless, the update may be done asap, but the python3 build should succeed then.

@pSub You did the update which is reverted by this PR.

matthiasbeyer commented:
I'm currently checking whether `vdirsyncer`, `khard` and `khal` still build after the update-revert.

matthiasbeyer commented:
It seems that they all build.

FRidh commented:
It seems a bunch of tests fail because they need network access. You can disable the specific tests (often you can use a special pytest option for that). Or, simply disable all tests. The former is preferred.

matthiasbeyer commented:
How'd you know its because they need network access? Anyways, yes, I will investigate and fixup this PR.

matthiasbeyer committed:
betamax: Disable tests

As said in #15037, the tests fail due to a network error as the tests
expect to have network access.

This simply disables the tests, which is not optimal, of course. I do
not understand the whole python infrastructure, so I'm not able to
disable only the failing-because-of-network-access tests.

matthiasbeyer commented:
@FRidh I disabled all the tests, as I do not understand how to disable only the `integration` tests, which are the failing ones.

I'm sorry. If someone can come up with a better fix, please go ahead.

---

Either way: Why was unstable released with this update included if the tests fail? I mean, seriousely, this breaks updates!

joachifm commented:
@matthiasbeyer only some builds are considered release critical, other failures will not block the channel.
